go/vt/vtgate/planbuilder/operators/query_planning.go

```diff
@@ -74,6 +74,13 @@ func runRewriters(ctx *plancontext.PlanningContext, root Operator) Operator {
 			return pushOrExpandHorizon(ctx, in)
 		case *Join:
 			return optimizeJoin(ctx, in)
+		case *ApplyJoin:
+			jp := collectOriginalExprs(in.JoinPredicates)
+			newPlan := mergeJoinInputs(ctx, in.LHS, in.RHS, jp, newJoinMerge(jp, in.JoinType))
+			if newPlan != nil {
+				return newPlan, Rewrote("merge routes of apply join into single operator")
+			}
```

Comment:
This is the new code we introduced to merge more routes that are possible after predicate pushing.